src/BlazorUI/Bit.BlazorUI.Tests/Checkboxes/BitCheckboxTests.cs

```diff
@@ -27,11 +27,15 @@ public class BitCheckboxTests : BunitTestContext
     ]
     public void BitCheckboxOnClickShouldWorkIfIsEnabled(Visual visual, bool defaultValue, bool isEnabled)
     {
-        var component = RenderComponent<BitCheckboxTest>(parameters =>
+        var clicked = false;
+        var changed = false;
+        var component = RenderComponent<BitCheckbox>(parameters =>
         {
             parameters.Add(p => p.Visual, visual);
```

Comment:
I think you should use `AddCascadingValue` instead of `Add` to send data to the `Visual` parameter. The `Visual` is cascading parameter.

Reply:
Actually it doesn't matter in that case and `Add` will call `AddCascading` see [here](https://github.com/bUnit-dev/bUnit/blob/main/src/bunit.core/ComponentParameterCollectionBuilder.cs#L49). bUnit does detect if it is a cascading one or not.

See it like that: From your component under test it does not matter, as we set a property. If the property has attribute A or B we don't care.
Still the question is: Why do we care then? We still have that switch, plus we have `AddCascadingValue`. The reason is simple:
```razor
<CascadingValue ...>
    <ComponentUnderTest>
        <ChildComponent>
        </ChildComponent>
    </ComponentUnderTest>
</CascadingValue>
```

Now if `ChildComponent` uses the cascading parameter but `ComponentUnderTest` we need a way to create those. And that is why you have `AddCascadingValue`.

To recap: If your component uses the cascading value directly and you have no sibling relying on that...you can use directly `Add`. If a sibling needs the parameter, then you have to use `AddCascadingValue`.

So in this case, it is preference. Sure I can change to `AddCascadingValue` if you want. Your choice ;)

Reply:
yeah, you're right. from the component pov it doesn't matter that the parameter is coming from what source.
but for the sake of compatibility/consistency and clarity, let's keep the cascading parameters being set by the `AddCascadingValue` method.

Reply:
I changed it to `AddCascadingValue`. Predominantly you use `Add` instead of `AddCascadingValue` in your tests. I could align that in a follow up if wished.